exasol_transformers_extension/udfs/models/filling_mask_udf.py

```diff
@@ -60,22 +60,19 @@ def get_batched_predictions(self, batch_df: pd.DataFrame) -> pd.DataFrame:
                 (batch_df['bucketfs_conn'] == bucketfs_conn) &
                 (batch_df['sub_dir'] == sub_dir)]
 
-            top_k = model_df['top_k'].iloc[0]
-            if not dataframe_operations.check_all_values_equal(
-                    model_df['top_k']):
-                raise Exception(
-                    "Inputs with the same model_name & bucketfs_conn & sub_dir "
-                    "values must have the same top_k values")
-
             current_model_key = (bucketfs_conn, sub_dir, model_name)
             if self.last_loaded_model_key != current_model_key:
                 self.set_cache_dir(model_df)
                 self.clear_device_memory()
-                self.load_models(model_name, top_k=top_k)
+                self.load_models(model_name)
                 self.last_loaded_model_key = current_model_key
 
-            model_pred_df = self.get_prediction(model_df)
-            result_df_list.append(model_pred_df)
+            unique_top_k_values = model_df['top_k'].unique()
+            for top_k in unique_top_k_values:
+                model_by_top_k_df = model_df[model_df['top_k'] == top_k]
+                self.setup_pipeline(top_k=top_k)
```

Comment:
Creating the pipeline after loading the models is not time consuming. 
I'm not sure how useful the different topk value for each input is, but I think it's good that we provide this flexibility.

Reply:
My question target into the direction, if it would be beneficial to create the pipeline only if the topk for a model changes and this also over batches. So, for example, 
```
Batch1:

model1, topk1
model1, topk1
model1, topk2
model1, topk2

Batch2:
model1, topk2
model1, topk2
model1, topk2
model1, topk2
```

Currently, we would create the pipeline twice for the combination model1, topk2, because we don't remember the model, topk combination from the last batch.

Reply:
you are right, it would be unnecessary to load each batch in this case